symbolic-base/src/ZkFold/Base/Protocol/Protostar/AccumulatorScheme.hs

```diff
@@ -8,90 +8,100 @@
 import           ZkFold.Base.Protocol.Protostar.CommitOpen   (CommitOpen (..))
-import           ZkFold.Base.Protocol.Protostar.FiatShamir   (FiatShamir (..))
-import           ZkFold.Base.Protocol.Protostar.NARK         (InstanceProofPair (..), NARKProof (..))
+import           ZkFold.Base.Protocol.Protostar.FiatShamir   (FiatShamir (..), transcriptFiatShamir)
+import           ZkFold.Base.Protocol.Protostar.NARK         (NARKInstanceProof (..), NARKProof (..))
 import           ZkFold.Base.Protocol.Protostar.Oracle       (RandomOracle (..))
 
 -- | Accumulator scheme for V_NARK as described in Chapter 3.4 of the Protostar paper
 --
 -- TODO: define the initial accumulator
 --
-class AccumulatorScheme pi f c m a where
+class AccumulatorScheme f i m c d k a where
   prover   :: a
-           -> Accumulator pi f c m        -- accumulator
-           -> InstanceProofPair pi c m    -- instance-proof pair (pi, π)
-           -> (Accumulator pi f c m, [c]) -- updated accumulator and accumulation proof
+           -> Accumulator f i m c k                      -- accumulator
+           -> NARKInstanceProof f i m c k                -- instance-proof pair (pi, π)
+           -> (Accumulator f i m c k, Vector (d - 1) c)  -- updated accumulator and accumulation proof
 
-  verifier :: pi                          -- Public input
-           -> [c]                         -- NARK proof π.x
-           -> AccumulatorInstance pi f c  -- accumulator instance acc.x
-           -> AccumulatorInstance pi f c  -- updated accumulator instance acc'.x
-           -> [c]                         -- accumulation proof E_j
-           -> (f, pi, [f], [c], c)        -- returns zeros if the accumulation proof is correct
+  verifier :: i f                                        -- Public input
+           -> Vector k c                                 -- NARK proof π.x
+           -> AccumulatorInstance f i c k                -- accumulator instance acc.x
+           -> AccumulatorInstance f i c k                -- updated accumulator instance acc'.x
+           -> Vector (d - 1) c                           -- accumulation proof E_j
+           -> (f, i f, Vector (k-1) f, Vector k c, c)    -- returns zeros if the accumulation proof is correct
 
   decider  :: a
-           -> Accumulator pi f c m        -- final accumulator
-           -> ([c], c)                    -- returns zeros if the final accumulator is valid
+           -> Accumulator f i m c k                      -- final accumulator
+           -> (Vector k c, c)                            -- returns zeros if the final accumulator is valid
 
 instance
-    ( Scale f c
-    , RandomOracle pi f         -- Random oracle for compressing public input
-    , RandomOracle c f          -- Random oracle ρ_NARK
+    ( AlgebraicMap f i d a
+    , AlgebraicMap (PU.PolyVec f (d + 1)) i d a
+    , Ring f
+    , Zip i
+    , KnownNat (d - 1)
+    , KnownNat (d + 1)
+    , Scale f c
     , HomomorphicCommit [f] c
```

Comment:
Is this still required when everything is on Vectors?

Reply:
Well, it does not compile without those constraints